docker/dockerfile

```diff
@@ -5,27 +5,32 @@ ARG USERID
 ARG USER
 
 # Setup environment
-ENV TERM linux
-ENV DEBIAN_FRONTEND noninteractive
+ENV TERM=linux
+ENV DEBIAN_FRONTEND=noninteractive
 RUN echo 'debconf debconf/frontend select Noninteractive' | debconf-set-selections
 
 # Copy requirement files and install dependencies
 COPY docker/requirements.txt .
 RUN apt-get update && apt-get install --no-install-recommends -y $(cat requirements.txt)
 RUN rm requirements.txt
 
-# Install Gazebo Fortress
-RUN sudo sh -c 'echo "deb http://packages.osrfoundation.org/gazebo/ubuntu-stable `lsb_release -cs` main" > /etc/apt/sources.list.d/gazebo-stable.list'
-RUN wget http://packages.osrfoundation.org/gazebo.key -O - | sudo apt-key add -
-RUN sudo apt-get update
-RUN sudo apt-get install ignition-fortress -y
+# Add the Gazebo repository and install Gazebo Harmonic
+RUN sudo sh -c 'echo "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/pkgs-osrf-archive-keyring.gpg] http://packages.osrfoundation.org/gazebo/ubuntu-stable $(lsb_release -cs) main" > /etc/apt/sources.list.d/gazebo-stable.list' && \
+    wget https://packages.osrfoundation.org/gazebo.gpg -O /usr/share/keyrings/pkgs-osrf-archive-keyring.gpg && \
+    sudo apt update && \
+    sudo apt install -y gz-harmonic && \
+    sudo apt remove -y ros-humble-ros-gz* && \
+    sudo apt autoremove -y && \
```

Comment:
Doing  ` sudo apt install -y gz-harmonic` should not bring `ros-gz`.
Is it just in case it is installed by another dependency?

Reply:
It's because on some other tests I did (on a custom workspace I have), I found out that with the full Humble installation some of the default packages for Gazebo were installed, so JIC (I can try it out without it, of course)

Reply:
Allright looks good

Reply:
I have tried it out without it JIC, and as expected, it's failing due to some libraries already installed with an older version:

```
108.9 The following packages have unmet dependencies:
108.9  ros-humble-ros-gzharmonic-sim : Conflicts: ros-humble-ros-gz-sim but 0.244.13-1jammy.20240217.080145 is to be installed
108.9 E: Unable to correct problems, you have held broken packages.
```